source/common/http/conn_manager_impl.cc

```diff
@@ -2235,6 +2239,10 @@ bool ConnectionManagerImpl::ActiveStreamDecoderFilter::recreateStream() {
   parent_.connection_manager_.doEndStream(this->parent_);
 
   StreamDecoder& new_stream = parent_.connection_manager_.newStream(*response_encoder, true);
+  (*parent_.connection_manager_.streams_.begin())->stream_info_.filter_state_ =
```

Comment:
related to my lazy init comment above, maybe only recreate this if there is actually data to move over? (In general we really try to avoid unnecessary allocations in the hot path)